File: engine/model/src/main/scala/tools/packet/field/effect/EnvironmentChange.scala
package tools.packet.field.effect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class EnvironmentChange(private var _env: String, private var _mode: Int) extends PacketInput {
  def env: String = _env

  def mode: Int = _mode

  override def opcode(): SendOpcode = SendOpcode.FIELD_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/field/effect/EnvironmentChange.java
package tools.packet.field.effect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record EnvironmentChange(String env, Integer mode) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FIELD_EFFECT;
   }
}